### ISSUE TITLE
Trigger mis-detects change to null_resource 'archive'

### ISSUE THREAD

**Issue by alexmbird** [terraform-aws-modules/terraform-aws-notify-slack#232]
## Description

I run Terraform through a GitLab pipeline, which checks out a fresh copy of the codebase and your module every time.

Each time I plan/apply, the resource is detected as changed and needs to be replaced:

```
Terraform will perform the following actions:
  # module.development.module.notify_slack.module.lambda.null_resource.archive[0] must be replaced
-/+ resource "null_resource" "archive" {
      ~ id       = "1648334273508296510" -> (known after apply)
      ~ triggers = { # forces replacement
          ~ "timestamp" = "1724922968954459000" -> "1724924530074214000"
            # (1 unchanged element hidden)
        }
    }
Plan: 1 to add, 0 to change, 1 to destroy.
```

This doesn't do any harm, but it adds churn in our Terraform output / workflow.

I think it's happening because you're triggering updates based on the timestamp of a file instead of a checksum.  Would it be possible to switch to checksums, either by default or as an option?

- [X] ✋ I have searched the open/closed issues and my issue is not listed.

## Versions

- Module version [Required]: 6.4.0

- Terraform version: 1.4.6
- Provider version(s):
  ```
  $ terraform providers -version
  + provider registry.terraform.io/hashicorp/aws v5.43.0
  + provider registry.terraform.io/hashicorp/local v2.5.1
  + provider registry.terraform.io/hashicorp/null v3.2.2
  + provider registry.terraform.io/hashicorp/random v3.6.1
  ```

## Reproduction Code
n/a

- Create notify_slack instance as normal
- `terraform apply`
- Delete checkout of your Terraform tree and get a new one
- `terraform plan` <-- wants to replace `null_resource.archive` because timestamp has changed

## Expected behavior

Once created, all resources remain stable

## Actual behavior

timestamp repeatedly triggers replacement of null_resource

**Comment by antonbabenko** [comment]
This issue has been resolved in version 6.5.0 :tada:

**Comment by alexmbird** [comment]
Thanks @antonbabenko - just updated and I can confirm it's fixed.
